modules/transport-netty4/src/test/resources/README.txt

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+#
+# This is README describes how the certificates in this directory were created.
+# This file can also be executed as a script
+#
+
+# 1. Create certificate key
+
+openssl req  -x509  -sha256  -newkey rsa:2048  -keyout certificate.key  -out certificate.crt  -days 1024  -nodes
```

Comment:
FYI The security plugin has classes that can generate certs for testing so there is no chance that 3 years down the line the certs that's checked in expires and needs to be updated in a future commit: https://github.com/opensearch-project/security/blob/main/src/integrationTest/java/org/opensearch/test/framework/certificate/CertificatesIssuer.java#L107

Reply:
Thank you, great to know!

Reply:
@reta are you going to integrate this suggestion into this PR or maybe do it as a follow up?

Reply:
> @reta are you going to integrate this suggestion into this PR or maybe do it as a follow up?

I will give it a try, thanks @andrross & @cwperks !

Reply:
Oops, quick update folks - the code in question heavily uses Bouncycastle which I am reluctant to bring into the netty4 transport, even as the test code (in general, I wanted to make it a part of test-framework, but Bouncycastle is a bummer), I think the approach is viable but using JCE / JDK would be preferred (needs some work though)

Reply:
> Hey, @reta how come you don't want to use BouncyCastle?

Hey @cwperks , so there are few things,  I prefer not to use BouncyCastle because:
 - it is not really needed, having regenerated certs for 10 years is a looong way to go
 - it will leak with the `test-framework` dependency all over the place (plugins, opensearch-java, etc) but at the moment, it is not essential part of OpenSearch (in contrast to security plugin)

>  According to this stackoverflow post, sun.security is not supported anymore: https://stackoverflow.com/questions/1615871/creating-an-x509-certificate-in-java-without-bouncycastle?rq=4

Correct, so this is why `sun.security.x509.*;` & co were not used

> What prevents us from using BouncyCastle for creating the certs following something like: https://medium.com/@lbroudoux/generate-self-signed-certificates-in-pure-java-83d3ad94b75?

Additional dependency like BC for quite a niche need seems to be too much (at least, to me).

Reply:
Hi @reta, 

I have some confusion on your reasoning here. 

> it is not really needed, having regenerated certs for 10 years is a looong way to go

Is the reasoning here just that 10 years is a long time and good enough? I am not sure that it being a long time means we should not use a dynamically generated cert when we have the means to do so. 

>  will leak with the test-framework dependency all over the place (plugins, opensearch-java, etc) but at the moment, it is not essential part of OpenSearch (in contrast to security plugin)

For this, I am not sure I follow since it seemed here: https://github.com/opensearch-project/OpenSearch/pull/12435#discussion_r1526754921 you were fine with the test-framework just not the Bouncycastle aspect? Maybe I am misunderstanding? What do you see as the issue with having the test-framework? 

It seems like you already realized that we cannot use the legacy packages for the generation so Bouncycastle is the best option unless we want to manually implement our own serialization. I know you feel that Bouncycastle is niche, but it seems like this would be a worthwhile addition. There are lots of dependencies in Opensearch and being able to get rid of hardcoded credentials seems pretty worthwhile. What do you think?

Reply:
Hey @scrawfor99 

> Is the reasoning here just that 10 years is a long time and good enough? I am not sure that it being a long time means we should not use a dynamically generated cert when we have the means to do so.

Many things will change, as always we need to have a precedent first and invest into solving it. In this particular case I took the simplest approach possible.

> For this, I am not sure I follow since it seemed here: https://github.com/opensearch-project/OpenSearch/pull/12435#discussion_r1526754921 you were fine with the test-framework just not the Bouncycastle aspect?

Correct, as ` test-framework` is used way outside OpenSearch core, additional dependency (like BC) have to be justified - if it brings benefits to others, we certainly could consider adding it. 

>  I know you feel that Bouncycastle is niche, but it seems like this would be a worthwhile addition. There are lots of dependencies in Opensearch and being able to get rid of hardcoded credentials seems pretty worthwhile. What do you think?

Certainly, if we have larger problem - please feel free to suggest the solution, if BC is the answer - let's consider it, no objections.